File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        if (n==1) return true;

        int pos = 1;
        while (pos<n && nums[pos]>=nums[pos-1]) {
            pos++ ;
        }
        if(pos == n) return true;
        if(nums[pos] > nums[0]) return false;
        if(pos==n-1) return true;

        pos++;
        while(pos<n && nums[pos]<=nums[0] && nums[pos]>=nums[pos-1]) {
            pos++ ;
        }

        return pos==n ;
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        if(n==1) return true;

        int pos = 1;
        while(pos<n && nums[pos] >= nums[pos-1] ) {
            pos = pos + 1;
        } 

        if (pos == n) return true;
        if (nums[pos] > nums[0] ) return false;
        

        pos = pos + 1;

        while(pos<n && nums[pos] <= nums[0] && nums[pos]>= nums[pos-1]) {
            pos = pos + 1;
        }
        return pos == n;
    }
}